### PR TITLE
feat(core): Convert saml controller to decorator style

### PR DESCRIPTION
**Pull request by flipswitchingmonkey**
Github issue / Community forum post (link here to close automatically):

### CODE REVIEW

File: packages/cli/src/sso/saml/routes/saml.controller.ee.ts [hunk 1]
```diff
@@ -1,41 +1,50 @@
...
-	async (req: SamlConfiguration.Update, res: express.Response) => {
+	/**
+	 * POST /sso/saml/config
+	 * Set SAML config
+	 */
+	@Post(SamlUrls.config, { middlewares: [samlLicensedOwnerMiddleware] })
+	async configPost(req: SamlConfiguration.Update, res: express.Response) {
 		const validationResult = await validate(req.body);
 		if (validationResult.length === 0) {
 			const result = await SamlService.getInstance().setSamlPreferences(req.body);
```
Comment (netroy) [inline packages/cli/src/sso/saml/routes/saml.controller.ee.ts:50]:
```suggestion
			const result = await this.samlService.setSamlPreferences(req.body);
```

File: packages/cli/src/sso/saml/routes/saml.controller.ee.ts [hunk 2]
```diff
@@ -46,33 +55,27 @@ samlControllerProtected.post(
...
+	/**
+	 * POST /sso/saml/config/toggle
+	 * Set SAML config
+	 */
+	@Post(SamlUrls.configToggleEnabled, { middlewares: [samlLicensedOwnerMiddleware] })
+	async toggleEnabledPost(req: SamlConfiguration.Toggle, res: express.Response) {
 		if (req.body.loginEnabled === undefined) {
 			throw new BadRequestError('Body should contain a boolean "loginEnabled" property');
 		}
 		await SamlService.getInstance().setSamlPreferences({ loginEnabled: req.body.loginEnabled });
```
Comment (netroy) [inline packages/cli/src/sso/saml/routes/saml.controller.ee.ts:69]:
```suggestion
		await this.samlService.setSamlPreferences({ loginEnabled: req.body.loginEnabled });
```

File: packages/cli/src/sso/saml/routes/saml.controller.ee.ts [hunk 3]
```diff
@@ -46,33 +55,27 @@ samlControllerProtected.post(
...
-	SamlUrls.acs,
-	samlLicensedAndEnabledMiddleware,
-	async (req: express.Request, res: express.Response) => {
+	/**
+	 * GET /sso/saml/acs
+	 * Assertion Consumer Service endpoint
+	 */
+	@Get(SamlUrls.acs, { middlewares: [samlLicensedAndEnabledMiddleware] })
+	async acsGet(req: express.Request, res: express.Response) {
 		const loginResult = await SamlService.getInstance().handleSamlLogin(req, 'redirect');
```
Comment (netroy) [inline packages/cli/src/sso/saml/routes/saml.controller.ee.ts:79]:
```suggestion
		const loginResult = await this.samlService.handleSamlLogin(req, 'redirect');
```

File: packages/cli/src/sso/saml/routes/saml.controller.ee.ts [hunk 4]
```diff
@@ -85,17 +88,14 @@ samlControllerProtected.get(
...
-	SamlUrls.acs,
-	samlLicensedAndEnabledMiddleware,
-	async (req: express.Request, res: express.Response) => {
+	/**
+	 * POST /sso/saml/acs
+	 * Assertion Consumer Service endpoint
+	 */
+	@Post(SamlUrls.acs, { middlewares: [samlLicensedAndEnabledMiddleware] })
+	async acsPost(req: express.Request, res: express.Response) {
 		const loginResult = await SamlService.getInstance().handleSamlLogin(req, 'post');
```
Comment (netroy) [inline packages/cli/src/sso/saml/routes/saml.controller.ee.ts:99]:
```suggestion
		const loginResult = await this.samlService.handleSamlLogin(req, 'post');
```

File: packages/cli/src/sso/saml/routes/saml.controller.ee.ts [hunk 5]
```diff
@@ -108,17 +108,14 @@ samlControllerProtected.post(
...
-	SamlUrls.initSSO,
-	samlLicensedAndEnabledMiddleware,
-	async (req: express.Request, res: express.Response) => {
+	/**
+	 * GET /sso/saml/initsso
+	 * Access URL for implementing SP-init SSO
+	 */
+	@Get(SamlUrls.initSSO, { middlewares: [samlLicensedAndEnabledMiddleware] })
+	async initSsoGet(req: express.Request, res: express.Response) {
 		const result = SamlService.getInstance().getLoginRequestUrl();
```
Comment (netroy) [inline packages/cli/src/sso/saml/routes/saml.controller.ee.ts:119]:
```suggestion
		const result = this.samlService.getLoginRequestUrl();
```

File: packages/cli/src/sso/saml/routes/saml.controller.ee.ts [hunk 6]
```diff
@@ -1,41 +1,50 @@
...
- * GET /sso/saml/config
- * Return SAML config
- */
-samlControllerProtected.get(
-	SamlUrls.config,
-	samlLicensedOwnerMiddleware,
-	(req: SamlConfiguration.Read, res: express.Response) => {
+	constructor() {
+		this.samlService = SamlService.getInstance();
+	}
```
Comment (netroy) [inline packages/cli/src/sso/saml/routes/saml.controller.ee.ts:23]: Please change this to 
```suggestion
	constructor(private samlService: SamlService) {}
```

and pass the service instance from the constructor call. That makes mocking a lot simpler.

**Review by netroy — CHANGES_REQUESTED** [review]

File: packages/cli/src/sso/saml/routes/saml.controller.ee.ts (hunk 6 above)
Comment (flipswitchingmonkey) [reply packages/cli/src/sso/saml/routes/saml.controller.ee.ts]: Done

**Review by netroy — APPROVED** [review]

**Pull request merged**